src/loaders/svg/tvgSvgSceneBuilder.cpp

```diff
@@ -807,12 +807,22 @@ static Scene* _useBuildHelper(SvgLoaderData& loaderData, const SvgNode* node, co
             }
             viewBoxClip->transform(mClipTransform);
 
-            scene->clip(viewBoxClip);
+            auto clippingLayer = Scene::gen();
+            clippingLayer->clip(viewBoxClip);
+            clippingLayer->push(scene);
+            return clippingLayer;
         }
-    } else {
-        scene->transform(mUseTransform);
+        return scene;
+    }
+
+    if (auto clipper = PAINT(scene)->clipper) {
+        auto& clipTransform = clipper->transform();
+        Matrix inv;
+        if (node->transform && inverse(node->transform, &inv)) clipTransform = inv * clipTransform;
```

Comment:
I think it's better to move `inv` inside the `if` since it won't be used if there's no `node->transform`.